deployments/azure-webapps-java-jar.yml

```diff
@@ -0,0 +1,72 @@
+# This workflow will build and push a node.js application to an Azure Web App when a commit is pushed to your default branch.
+#
+# This workflow assumes you have already created the target Azure App Service web app.
+# For instructions see https://docs.microsoft.com/en-us/azure/app-service/quickstart-nodejs?tabs=linux&pivots=development-environment-cli
+#
+# To configure this workflow:
+#
+# 1. Download the Publish Profile for your Azure Web App. You can download this file from the Overview page of your Web App in the Azure Portal.
+#    For more information: https://docs.microsoft.com/en-us/azure/app-service/deploy-github-actions?tabs=applevel#generate-deployment-credentials
+#
+# 2. Create a secret in your repository named AZURE_WEBAPP_PUBLISH_PROFILE, paste the publish profile contents as the value of the secret.
+#    For instructions on obtaining the publish profile see: https://docs.microsoft.com/azure/app-service/deploy-github-actions#configure-the-github-secret
+#
+# 3. Change the value for the AZURE_WEBAPP_NAME. Optionally, change the JAVA_VERSION environment variable below.
+#
+# For more information on GitHub Actions for Azure: https://github.com/Azure/Actions
+# For more information on the Azure Web Apps Deploy action: https://github.com/Azure/webapps-deploy
+# For more samples to get started with GitHub Action workflows to deploy to Azure: https://github.com/Azure/actions-workflow-samples
+
+name: Build and deploy JAR app to Azure Web App
+
+env:
+  AZURE_WEBAPP_NAME: your-app-name    # set this to the name of your Azure Web App
+  JAVA_VERSION: '11'                  # set this to the Java version to use
+
+on:
+  push:
+    branches:
+      - $default-branch
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Java version
+        uses: actions/setup-java@v2.3.1
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
```

Comment:
We should enable caching.

Reply:
@jeremyepling Good _cache_! 😉 

Do you want to enable caching for the other languages that don't support caching in the `setup-xxx` actions? On one hand it'll improve perf, but now sure if you want to expose all that in *starter* workflows.

```yaml
- name: Cache multiple paths
  uses: actions/cache@v2
  with:
    path: |
      ~/cache
      !~/cache/exclude
    key: ${{ runner.os }}-${{ hashFiles('**/lockfiles') }}
 ```

Reply:
@JasonFreeberg starter workflows set a guiding example for user to create good workflows. Hence it would be okay to use the `actions/cache` action for such languages.
@pulkitaggarwl any thoughts?

Reply:
@bishal-pdMSFT  this makes sense

Reply:
@JasonFreeberg` actions/setup-<language>` support in-built caching eg. https://github.com/actions/setup-node#caching-packages-dependencies So, I think you can use the setup-<language> action for caching dependencies. And if setup action does support cache then you use `actions/cache` for caching.

Reply:
Just pushed a commit to add caching for .NET, Python, PHP, and Node workflows.